src/main/java/org/opentripplanner/gtfs/TripPatternCache.java

```diff
@@ -0,0 +1,112 @@
+package org.opentripplanner.gtfs;
+
+import com.google.common.collect.ArrayListMultimap;
+import com.google.common.collect.Multimap;
+import java.util.HashMap;
+import java.util.Map;
+import javax.annotation.Nonnull;
+import org.opentripplanner.transit.model.framework.FeedScopedId;
+import org.opentripplanner.transit.model.network.Route;
+import org.opentripplanner.transit.model.network.StopPattern;
+import org.opentripplanner.transit.model.network.TripPattern;
+import org.opentripplanner.transit.model.timetable.Direction;
+import org.opentripplanner.transit.model.timetable.Trip;
+
+/**
+ * A synchronized cache of trip patterns that are added to the graph.
+ */
+public class TripPatternCache {
```

Comment:
@t2gran Might have something to say about the location of this code.

Reply:
I'd like to have a shared cache between SIRI and GTFS-RT

Reply:
Is this used locally inside the GenerateTripPatternsOperation, so this is located in the right place. Unsure if this is a `Cache` or just an `Index` (current usage - import). We probably want something in the RT updates as well, but since we are implementing this in a totally new way in the transit model refactoring - doing it in the RT part now will be work we will throw away. This is totally redesigned in the new model.

PS! it should be package local - class and methods.